### PR TITLE
Immediately decidable property

### PR DESCRIPTION
**Pull request by tkmct**

### CODE REVIEW

File: contracts/Predicate/CompiledPredicate.sol [hunk 1]
```diff
@@ -16,4 +16,8 @@ interface CompiledPredicate {
         returns (bool);
     function decideTrue(bytes[] calldata _inputs, bytes[] calldata _witness)
         external;
+    function decideWithWitness(
```
Comment (syuhei176) [inline contracts/Predicate/CompiledPredicate.sol:19]: note: we should remove "decideTrue" and modify solidity-generator to change "decideTrue" to "decideWithWitness".

File: contracts/Predicate/ThereExistsSuchThatQuantifier.sol [hunk 2]
```diff
@@ -42,4 +59,89 @@ contract ThereExistsSuchThatQuantifier is LogicalConnective {
...
+        for (uint256 i = 0; i < _witness.length - 1; i++) {
+            witness[i] = _witness[i + 1];
+        }
+        return predicate.decideWithWitness(property.inputs, witness);
+    }
+
+    /**
+     * @dev Replace placeholder by quantified in propertyBytes
+     */
+    function replaceVariable(
```
Comment (syuhei176) [inline contracts/Predicate/ThereExistsSuchThatQuantifier.sol:90]: Can we make Quantifier contract? For instance, ForAllSuchThatQuantifier and ThereExistsSuchThatQuantifier inherit Quantifier and Quantifier has `replaceVariable` method.

File: contracts/Predicate/CompiledPredicate.sol (hunk 1 above)
Comment (tkmct) [reply contracts/Predicate/CompiledPredicate.sol:19]: ah, okay. I'll do that

File: contracts/Predicate/ThereExistsSuchThatQuantifier.sol (hunk 2 above)
Comment (tkmct) [reply contracts/Predicate/ThereExistsSuchThatQuantifier.sol:90]: that's a good abstraction. I'll do that in this PR.

Comment (tkmct) [reply contracts/Predicate/ThereExistsSuchThatQuantifier.sol:90]: @syuhei176 quantifier needs to know both ForAll address and ThereExists address. Is that possible?

**Pull request merged**